gcovr/utils.py

```diff
@@ -69,6 +70,42 @@ def realpath(path):
     realpath = os.path.realpath
 
 
+@functools.lru_cache(maxsize=1)
+def is_fs_case_insensitive():
+    cwd = os.getcwd()
+    # Guessing if file system is case insensitive.
+    # The working directory is not the root and accessible in upper and lower case.
+    ret = (
+        (cwd != os.path.sep)
+        and os.path.exists(cwd.upper())
+        and os.path.exists(cwd.lower())
+    )
+    if ret:
+        logger.info("File system is case insensitive.")
```

Comment:
If this should be logged, then both cases and as debug message.
```suggestion
        logger.debug(f"File system is case {'insensitive' if ret else 'sensitive'}.")
```